prometheus/inc/class-plugin.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+namespace Automattic\VIP\Prometheus;
+
+use Prometheus\CollectorRegistry;
+use Prometheus\RegistryInterface;
+use Prometheus\RenderTextFormat;
+use Prometheus\Storage\Adapter;
+use Prometheus\Storage\APCng;
+use Prometheus\Storage\InMemory;
+use WP;
+use WP_Query;
+
+final class Plugin {
+	private static ?Plugin $instance = null;
+	private RegistryInterface $registry;
+
+	public static function get_instance(): self {
+		if ( ! self::$instance ) {
+			self::$instance = new self();
+		}
+
+		return self::$instance;
+	}
+
+	private function __construct() {
+		$this->registry = self::create_registry();
+
+		add_action( 'plugins_loaded', [ $this, 'plugins_loaded' ] );
+		add_action( 'init', [ $this, 'init' ] );
+	}
+
+	public function plugins_loaded(): void {
+		$collectors = apply_filters( 'vip_prometheus_collectors', [] );
+		if ( ! is_array( $collectors ) ) {
+			$collectors = [];
+		}
+
+		$collectors = array_filter( $collectors, fn ( $collector) => is_object( $collector ) && $collector instanceof CollectorInterface );
```

Comment:
Any particular reason to use array_filter + array_map vs a basic foreach?

Reply:
No :-) Two lines instead of 6-7.

Reply:
Rewrote the code.